### PR TITLE
[JN-1088] Admin form UX improvements + new task type

### PR DESCRIPTION
**Pull request by MatthewBemis**
#### DESCRIPTION (include screenshots, and mobile screenshots for participant UX)

From talking with Erin and based on feedback from demos, it feels like the initial v0 UX for admin form editing was trying to do two separate things:

1) Allow admins to edit or correct responses on behalf of users (not likely a super common use-case, but still a necessary feature)
2) Allow study staff to do long or short-form data entry about a participant

As a result, this PR breaks the UX into two distinct flavors. To solve for the first use-case, participant-facing forms have the option to additionally be completed/edited by study-staff. For the second use-case, study staff have the option to create a "study staff" form which has the new survey type `ADMIN`. These are displayed separately from the research surveys so they don't live in the same bucket of forms as the participant-facing forms. ADMIN forms have ADMIN tasks associated with them, which do not show up on the participant dashboard.

This PR also adds an autosave indicator and adds an improved toggle for entering view/edit/print modes.

Screenshot 2024-06-04 at 8 57 03 AM

Screenshot 2024-06-04 at 2 13 13 PM

Screenshot 2024-06-04 at 9 00 43 AM

Screenshot 2024-06-04 at 9 00 33 AM

Screenshot 2024-06-04 at 9 12 06 AM

#### TO TEST:  *(simple manual steps for confirming core behavior -- used for pre-release checks)*

* Restart AdminApi and ParticipantApi
* Add a new Study Staff form and assign it to existing participants https://localhost:3000/ourhealth/studies/ourheart/env/sandbox/forms
* Log in to the participant site as "basic@test.com" and verify that you do not see the admin form
* Log back in as an admin, check that this task shows up for this enrollee and verify that you can complete it
* Verify that you can still edit and print existing research surveys (assuming they are marked as editable by admins)

### CODE REVIEW

File: ui-admin/src/study/participants/participantList/ParticipantList.tsx [hunk 1]
```diff
@@ -100,14 +100,14 @@ function ParticipantList({ studyEnvContext }: {studyEnvContext: StudyEnvContextT
...
-    accessorKey: 'profile.familyName',
+    accessorFn: row => row.profile?.familyName,
     meta: {
       columnType: 'string'
     }
   }, {
     id: 'givenName',
     header: 'Given name',
-    accessorKey: 'profile.givenName',
+    accessorFn: row => row.profile?.givenName,
```
Comment (MatthewBemis) [inline ui-admin/src/study/participants/participantList/ParticipantList.tsx:107]: driveby fixes for browser console warning `console.js:273 "lastLogin" in deeply nested key "participantUser.lastLogin" returned undefined. `

File: ui-admin/src/study/participants/enrolleeView/EnrolleeView.tsx [hunk 2]
```diff
@@ -90,7 +93,7 @@ export function LoadedEnrolleeView({ enrollee, studyEnvContext, onUpdate }:
           <div style={navDivStyle}>
             <ul className="list-unstyled">
               <li style={navListItemStyle} className="ps-3">
-                <NavLink to="." className={getLinkCssClasses}>Overview</NavLink>
+                <NavLink end to="." className={getLinkCssClasses}>Overview</NavLink>
```
Comment (MatthewBemis) [inline ui-admin/src/study/participants/enrolleeView/EnrolleeView.tsx:111]: driveby fix for "Overview" navlink always having the "selected" CSS applied to it

File: ui-admin/src/study/participants/survey/PrintFormModal.tsx [hunk 3]
```diff
@@ -34,7 +34,8 @@ const PrintFormModal = ({ survey, answers, resumeData }: DownloadFormViewProps)
   return <Modal show={true} fullscreen={true}>
     <Modal.Body className="m-0 p-0">
       <div className="d-print-none d-flex justify-content-center py-2">
-        <Link className="btn btn-secondary" to="..">Done</Link>
+        {/* @ts-ignore Link to type also supports numbers for back operations */}
+        <Link className="btn btn-secondary" to={-1}>Done</Link>
```
Comment (connorlbark) [inline ui-admin/src/study/participants/survey/PrintFormModal.tsx:38]: curious: is there a behavior difference b/w '..' and -1?

**Review by connorlbark — COMMENTED** [review]
I ran into a couple strange bugs.

First, I wasn't able to save a response for a user when saving an admin form:

https://github.com/broadinstitute/juniper/assets/17440010/cff4e7fc-91c3-478c-96d7-83dc684bffa4

I can only get it to save if it I hit the `complete` button, but it doesn't save in progress surveys.

Second, I could not replicate it, but I somehow saved a new version of an admin form and it ended up as a research form. I tried creating new admin surveys and creating new versions, but none of them exhibited the same behavior. Very strange...

image

File: ui-admin/src/study/participants/survey/PrintFormModal.tsx (hunk 3 above)
Comment (MatthewBemis) [reply ui-admin/src/study/participants/survey/PrintFormModal.tsx:38]: `..` navigates up one level in the route, `-1` just goes to the previous page, wherever that may be. I had to change this because taskId is specified in the URL query params now, and `..` causes those params to be stripped out, whereas -1 preserves them. I could have also kept using `..` by grabbing the query params and passing them back in as a prop

Comment (connorlbark) [reply ui-admin/src/study/participants/survey/PrintFormModal.tsx:38]: Ahh, very interesting. Thanks

**Review by devonbush — APPROVED** [review]
Looks great -- really appreciate how you accomplished this with relatively little code.  As discussed, I'll handle any needed migrations for HeartHive

**Review by connorlbark — APPROVED** [review]
works great! super snappy with the in-progress being updated in real time 😮
